### PR TITLE
Implemented options -d/--data for $QW_DATA support

### PR DESCRIPTION
**Pull request by wdconinc**
QwEventBuffer is the only class using $QW_DATA env var support. Now we
have a static default data directory ("/adaq1/data1/apar") which is
overriden by QW_DATA or -d/--data.

QwOptions was slightly modified in its env var parsing. Instead of
requiring a prefix "Qw" (i.e. Qwrun=100 is equivalent to --run=100) it
now uses "QW_" as prefix.

Unfortunately boost::program_options::parse_environment does not allow
for the "allow_unregistered()" decorator, so we immediately trip up over
the unparsed $QW_FIELDMAP and the like. We bypass this problem by
writing a functor to which we can specify a set of lowercase appendages
to ignore (e.g. "fieldmap"). The rest the functor does it stripping the
prefix "QW_" and turning the rest to lowercase for parsing "QW_DATA" as
"data" and then recognizing as the argument of --data.

tl;dr
```
build/qwparity_simple -r 10 -d /my/data/path
QW_DATA=/your/data/path build/qwparity_simple -r 10 -d /my/data/path
QW_DATA=/my/data/path build/qwparity_simple -r 10
```
now all have the same result.

Notes on the PR:
- Needs testing on older boost versions. Not sure of the need for preprocessor guards based on the functor functionality. I think that's been there for long enough though.
- Might make sense to fold in QW_PRMINPUT parsing in QwParameterFile, but I remember that for some reason QwOptions and QwParameterFile could not depend on each other.

### CODE REVIEW

**Comment by wdconinc** [comment]
Two other natural candidates for a similar treatment are QW_TMP used in QwEPICSEvent, and QW_ROOTFILES in QwRootFile and QwPromptSummary. Then all those can be specified on the command line.

As for QW_PRMINPUT, if we specify these on the command line we might need to make sure that they can be used multiple times, e.g. --prminput <path1> --prminput <path2>

**Comment by wdconinc** [comment]
> Might make sense to fold in QW_PRMINPUT parsing in QwParameterFile, but I remember that for some reason QwOptions and QwParameterFile could not depend on each other.

The reason is that when you specify --add-config, processed by QwOptions, it actually loads that as a QwParameterFile so it can take advantage of the search path. I guess we want the following to work:
```
build/qwparity_simple --prminput Analysis/prminput --prminput Parity/prminput --add-config qwparity.conf
```
without any environment variables, or with QW_PRMINPUT set as well (maybe even to a colon-separated list of paths, pie in the sky).
